### PR TITLE
Tai changes stage 3.1

### PR DESCRIPTION
**Pull request by bohdan-oheruk-plvision**

### CODE REVIEW

File: stratum/hal/lib/phal/tai/tai_wrapper/BUILD [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        "tai_wrapper.h",
+        "tai_object.h",
+        "network_interface.h",
+        "host_interface.h",
+        "module.h",
+        "types_converter.h",
+        "tai_attribute.h",
+        "tai_wrapper_interface.h",
+    ],
+
```
Comment (Yi-Tseng) [inline stratum/hal/lib/phal/tai/tai_wrapper/BUILD]: remove empty line here

File: stratum/hal/lib/phal/tai/tai_wrapper/BUILD [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    deps = [
+        "//stratum/hal/lib/common:test_main",
+        "@com_github_google_glog//:glog",
+        "@com_google_googletest//:gtest",
+
```
Comment (Yi-Tseng) [inline stratum/hal/lib/phal/tai/tai_wrapper/BUILD]: remove empty line here.

File: stratum/hal/lib/phal/tai/tai_wrapper/test/tai_wrapper_mock.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "gmock/gmock.h"
+
+
+namespace stratum {
+namespace hal {
+namespace phal {
+namespace tai {
+
+class tai_wrapper_mock : public TAIWrapperInterface {
```
Comment (Yi-Tseng) [inline stratum/hal/lib/phal/tai/tai_wrapper/test/tai_wrapper_mock.h]: `TaiWrapperMock`

File: stratum/hal/lib/phal/tai/tai_wrapper/BUILD [hunk 4]
```diff
@@ -0,0 +1,111 @@
+#
+# Copyright 2018 Google LLC
+# Copyright 2018-present Open Networking Foundation
+#
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/BUILD]: Fix copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/host_interface.cc [hunk 5]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
+ *
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/host_interface.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/host_interface.h [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/host_interface.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/module.cc [hunk 7]
```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/module.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/module.h [hunk 8]
```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/module.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/network_interface.cc [hunk 9]
```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/network_interface.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/network_interface.h [hunk 10]
```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/network_interface.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_attribute.cc [hunk 11]
```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_attribute.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_attribute.h [hunk 12]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_attribute.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_manager.cc [hunk 13]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_manager.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_manager.h [hunk 14]
```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_manager.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_object.cc [hunk 15]
```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_object.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_object.h [hunk 16]
```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_object.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_wrapper.cc [hunk 17]
```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_wrapper.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_wrapper.h [hunk 18]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_wrapper.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_wrapper_interface.cc [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+
+bool TAIPathItem::isValid() const {
+  return !((object_type == tai_object_type_t::TAI_OBJECT_TYPE_NULL) ||
+           (object_type == tai_object_type_t::TAI_OBJECT_TYPE_MAX));
+}
+
+}  // namespace tai
+}  // namespace phal
+}  // namespace hal
+}  // namespace stratum
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_wrapper_interface.cc]: Interfaces have no .cc files

File: stratum/hal/lib/phal/tai/tai_wrapper/tai_wrapper_interface.h [hunk 20]
```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/tai_wrapper_interface.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/test/tai_interface_test.cc [hunk 21]
```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/test/tai_interface_test.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/test/tai_manager_test.cc [hunk 22]
```diff
@@ -0,0 +1,420 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/test/tai_manager_test.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/test/tai_object_mock.h [hunk 23]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/test/tai_object_mock.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/test/tai_test_manager_wrapper.h [hunk 24]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/test/tai_test_manager_wrapper.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/test/tai_wrapper_mock.h [hunk 25]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/test/tai_wrapper_mock.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/types_converter.cc [hunk 26]
```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/types_converter.cc]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/types_converter.h [hunk 27]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2018 Google LLC
+ * Copyright 2018-present Open Networking Foundation
```
Comment (pudelkoM) [inline stratum/hal/lib/phal/tai/tai_wrapper/types_converter.h]: copyright

File: stratum/hal/lib/phal/tai/tai_wrapper/BUILD (hunk 1 above)
Comment (maksym-tropets-plvision) [reply stratum/hal/lib/phal/tai/tai_wrapper/BUILD]: Done. Please review.

File: stratum/hal/lib/phal/tai/tai_wrapper/BUILD (hunk 2 above)
Comment (maksym-tropets-plvision) [reply stratum/hal/lib/phal/tai/tai_wrapper/BUILD]: Removed all empty lines between build arguments. Please review.

File: stratum/hal/lib/phal/tai/tai_wrapper/test/tai_wrapper_mock.h (hunk 3 above)
Comment (maksym-tropets-plvision) [reply stratum/hal/lib/phal/tai/tai_wrapper/test/tai_wrapper_mock.h]: Changed. Please review.

File: stratum/hal/lib/phal/tai/tai_wrapper/BUILD (hunk 4 above)
Comment (maksym-tropets-plvision) [reply stratum/hal/lib/phal/tai/tai_wrapper/BUILD]: Fixed copyright headers in all files (recursively) under the `stratum/hal/lib/phal/tai/` path. Please review.

**Pull request closed without merging**